h/static/scripts/directive/group-list.js

```diff
@@ -1,17 +1,64 @@
 'use strict';
 
+var assert = require('assert');
+
+function GroupsListController($scope) {
+  $scope.expandedGroupId = undefined;
+
+  // used to keep the dropdown from closing when the user
+  // interacts with the inline link share pane within the groups list
+  $scope.stopClickPropagation = function($event) {
+    $event.stopPropagation();
+  }
+
+  // show the share link for the specified group or clear it if
+  // null
+  $scope.toggleShareLink = function (groupId) {
+    if (!groupId || $scope.expandedGroupId === groupId) {
+      $scope.expandedGroupId = undefined;
+    } else {
+      $scope.expandedGroupId = groupId;
+    }
+  };
+
+  $scope.shouldShowShareLink = function (groupId) {
+    return $scope.expandedGroupId === groupId;
+  }
+
+  $scope.linkForGroup = function (groupId) {
+    return $scope.baseURI + 'groups/' + groupId;
+  }
+}
+
 /**
  * @ngdoc directive
  * @name groupList
  * @restrict AE
  * @description Displays a list of groups of which the user is a member.
  */
-
 // @ngInject
 module.exports = function (groups) {
   return {
-    link: function (scope, elem, attrs) {
-      scope.groups = groups;
+    controller: ['$scope', GroupsListController],
```

Comment:
If you add an `// @ngInject` comment before the definition of `GroupsListController`, the array-form function definition isn't necessary.